### PR TITLE
Remove workaround in InferenceData conversion; require ArviZ >=0.11.4

### PR DESCRIPTION
**Pull request by michaelosthege**
This PR removes a workaround in the InferenceData conversion and at the same time fixes a bug that was present in the workaround.

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes? → possibly ArviZ API changes
+ [x] important background, or details about the implementation → 
+ [x] are the changes—especially new features—covered by tests and docstrings? → yup
+ [x] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style) → aye

### CODE REVIEW

File: pymc/backends/arviz.py [hunk 1]
```diff
@@ -160,7 +120,6 @@ def __init__(
         model=None,
         save_warmup: Optional[bool] = None,
         density_dist_obs: bool = True,
```
Comment (OriolAbril) [inline pymc/backends/arviz.py:122]: doesn't need to be in this PR, but we can remove this argument. densitydist api will be fixed in v4 so this is no longer necessary.

**Review by OriolAbril — APPROVED** [review]

**Review by twiecki — APPROVED** [review]

**Comment by michaelosthege** [comment]
thanks for dealing with the dependencies, @OriolAbril !

I just rebased on main to resolve conflicts. Also I rearranged the commits such that the dependency upgrade is now all in one. Let's hope the CI likes it.

**Comment by michaelosthege** [comment]
@OriolAbril can you check the diff again? I had to mark a test as XFAIL because I couldn't reproduce the float64/float32 casting error locally.
My feeling is that this is not critical, and because it's a good thing to have less strict dependencies we can go ahead?

File: conda-envs/environment-dev-py37.yml [hunk 2]
```diff
@@ -13,18 +13,16 @@ dependencies:
...
 - numpydoc>=0.9
-- pandas=0.24
+- pandas>=0.24
 - pre-commit>=2.8.0
 - pydata-sphinx-theme
 - pytest-cov>=2.5
 - pytest>=3.0
 - python-graphviz
 - python=3.7
-- recommonmark>=0.4
```
Comment (twiecki) [inline conda-envs/environment-dev-py37.yml]: On purpose?

**Review by twiecki — APPROVED** [review]

File: conda-envs/environment-dev-py37.yml (hunk 2 above)
Comment (michaelosthege) [reply conda-envs/environment-dev-py37.yml]: I just undid that change while rebasing, since it was part of #5057 either.

**Comment by twiecki** [comment]
Can merge once we get green light.

**Comment by twiecki** [comment]
👍

**Pull request merged**